### PR TITLE
make gtiff emit merged consecutive multi-range reads

### PR DESCRIPTION
**Pull request by tbonfort**
addressing #1295

### CODE REVIEW

File: gdal/frmts/gtiff/geotiff.cpp [hunk 1]
```diff
@@ -4110,13 +4110,53 @@ void* GTiffRasterBand::CacheMultiRange( int nXOff, int nYOff,
                     apData.push_back(static_cast<GByte*>(pBufferedData) + nAccOffset);
                     nAccOffset += aOffsetSize[i].second;
                 }
+                int nMultiRet;
                 VSILFILE* fp = VSI_TIFFGetVSILFile(th);
-                if( VSIFReadMultiRangeL(
+
+                if( aOffsetSize.size() > 1 )
```
Comment (rouault) [inline gdal/frmts/gtiff/geotiff.cpp]: cannot this just be done in the loop a few line above ?

Comment (tbonfort) [reply gdal/frmts/gtiff/geotiff.cpp]: no, you need the first loop to construct the un-merged offsets/sizes, as those are the ones that get cached.

Comment (rouault) [reply gdal/frmts/gtiff/geotiff.cpp]: from a quick look at tifvsi.cpp, I don't see why VSI_TIFFSetCachedRanges() couldn't be called with the merged ranges

Comment (tbonfort) [reply gdal/frmts/gtiff/geotiff.cpp]: you are right, I've merged the two loops into one

**Pull request merged**